Data/models/Domain.swift

```diff
@@ -44,9 +44,14 @@ public final class Domain: NSManagedObject, CRUD {
         //  A solution to consider is creating a new background context here, creating, saving, and then re-fetching
         //   that object in the requested context (regardless if it is `viewContext` or not)
         var newDomain: Domain!
-        context.performAndWait {
-            newDomain = Domain(entity: Domain.entity(context), insertInto: context)
-            newDomain.url = domainString
+        // Since most calls are from main queue performandWait has no adavantage.
+//        context.performAndWait {
```

Comment:
Removing this will introduce a race condition regression. Without blocking here, if two places attempt to pull "google.com" at the same time, it will create two domains, since the store does not have one yet.

Reply:
I will revert it, but we need an overhaul of the Database manager. `performblockandwait` is not of advantage if calling queue and context are both main. And if the calling queue is main and the context is background then it will block main thread. Also nothing is in place to make sure two background threads don't create two background context and add domains, which may cause redundancy.

Reply:
Yeah, we really should serialize this in a more robust way. Is this going to cause problems for the implementation though? Or was the issue just the default context abstraction?

Reply:
The problem will be choppy main thread. But i guess we can ignore this till we refactor

Reply:
Agreed, thankfully these are extremely lightweight operations. But something to improve.